### PR TITLE
RavenDB-14225 Allow to define separate conflict revision configuration

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 1]
```diff
@@ -185,6 +185,9 @@ public void InitializeFromDatabaseRecord(DatabaseRecord dbRecord)
         {
             try
             {
+                if (dbRecord.RevisionsForConflicts != null)
+                    ConflictConfiguration.Default = dbRecord.RevisionsForConflicts;
```
Comment (ayende) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:189]: This is a _static_ instance, no?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:189]: yes! done.

**Comment by karmeli87** [comment]
Same as previously, first time in the `ctor` of `RevisionsStorage` and then updated by `InitializeFromDatabaseRecord`.

File: src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperation.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+using Raven.Client.Documents.Session;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Converters;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    public class EditRevisionsForConflictsOperation : IServerOperation<EditRevisionsForConflictsOperationResult>
```
Comment (arekpalinski) [inline src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperation.cs]: Shouldn't be renamed to `ConfigureRevisionsForConflictsOperation`? For consistency with `ConfigureRevisionsOperation`

File: src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperationResult.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Raven.Client.ServerWide.Operations
+{
+    public class EditRevisionsForConflictsOperationResult
```
Comment (arekpalinski) [inline src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperationResult.cs]: Rename to `ConfigureRevisionsForConflictsOperationResult`?

File: src/Raven.Server/Documents/Revisions/RevisionsStorage.cs [hunk 4]
```diff
@@ -343,6 +346,10 @@ public RevisionsCollectionConfiguration GetRevisionsConfiguration(string collect
             if (configuration == null)
                 configuration = GetRevisionsConfiguration(collectionName.Name);
 
+            if (configuration.Disabled && 
+                nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication) == false)
+                return false;
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:351]: How is that change related to introduced option of defining separate conflict revision configuration? Or this is another fix?

Comment (karmeli87) [reply src/Raven.Server/Documents/Revisions/RevisionsStorage.cs:351]: I guess it both.

On conflict we were passing the configuration directly, and if we don't check for `Disabled` we will keep conflicted revision even if they were disabled.
In general if the configuration is disabled we should simply leave the state as is.

File: src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperationResult.cs (hunk 3 above)
Comment (karmeli87) [reply src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperationResult.cs]: done

File: src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperation.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Client/ServerWide/Operations/EditRevisionsForConflictsOperation.cs]: done
